app/lib/actions/pulp/repository/presenters/puppet_presenter.rb

```diff
@@ -25,11 +25,14 @@ def humanized_details
           end
 
           def finished_count
-            task_result['finished_count']
+            return 0 unless task_progress_details.key?('modules')
+            task_progress_details['modules']['finished_count']
           end
 
           def total_count
-            task_progress_details['modules']['total_count'] || 0
+            return 0 unless task_progress_details.key?('modules')
+            return 0 if task_progress_details['modules']['total_count'].nil?
```

Comment:
I can go this route, I went with dual guard statements to capture the different nature of each and cause I thought it made it clear the two situations that can arise.